### PR TITLE
FW-4217 SiteWidget create/update/delete APIs

### PR DESCRIPTION
**Pull request by NolanVH**
### Description of Changes
This PR adds the SiteWidget create/update/delete APIs.

~As discussed, the visibility field will be set to match the site for now and we will update the APIs to allow visibility changes with specific permissions in a future PR.~

Additionally, this PR adds a unique_together setting to ensure that a SiteWidget is unique in a single SiteWidgetList (homepage or page widgets).

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- [x] Admin Panel has been updated if models have been added or modified
- [x] Migrations have been updated and committed if applicable
- [x] Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [x] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/migrations/0053_alter_sitewidgetlistorder_unique_together.py
Comment (bronson-c) [inline firstvoices/backend/migrations/0053_alter_sitewidgetlistorder_unique_together.py:1]: Just a heads-up, this migration will likely need to be merged before merging to main.

File: firstvoices/backend/tests/test_apis/test_widgets_api.py
Comment (bronson-c) [inline firstvoices/backend/tests/test_apis/test_widgets_api.py:1]: This file could probably use some more tests to check that the create/update/delete functions are working as expected. (Check that the db is updated as expected and that settings models are deleted as appropriate, etc.)

**Comment by sarahfirstvoices** [comment]
@NolanVH Is there a separate ticket for the visibility/permission updates?

File: firstvoices/backend/serializers/widget_serializers.py [hunk 1]
```diff
@@ -22,21 +22,31 @@ class Meta:
...
+    CreateSiteContentSerializerMixin,
+    SiteContentLinkedTitleSerializer,
+    WidgetDetailSerializer,
 ):
-    url = SiteHyperlinkedIdentityField(view_name="api:sitewidget-detail")
-    visibility = serializers.CharField(source="get_visibility_display")
+    url = SiteHyperlinkedIdentityField(
+        view_name="api:sitewidget-detail", read_only=True
+    )
+    visibility = serializers.CharField(source="get_visibility_display", read_only=True)
```
Comment (sarahfirstvoices) [inline firstvoices/backend/serializers/widget_serializers.py:49]: I'm not sure this needs any special permission handling... Based on the chart in our wiki, I thought Language Admins and Superadmins had full access to edit widgets, and nobody else had any edit access.

Comment (NolanVH) [reply firstvoices/backend/serializers/widget_serializers.py]: As discussed I have added the ability to edit visibility in this PR.

**Comment by NolanVH** [comment]
> @NolanVH Is there a separate ticket for the visibility/permission updates?

Yes, the ticket can be found here: [FW-4616](https://firstvoices.atlassian.net/browse/FW-4616).
As we discussed I have added in the ability to edit the visibility field to this PR as the widgets don't need separate permissions.

**Review by sarahfirstvoices — APPROVED** [review]

**Pull request merged**